artichoke-backend/src/hash.rs

```diff
@@ -0,0 +1,14 @@
+use crate::core::BuildHasher;
+use crate::error::Error;
+use std::collections::hash_map::RandomState;
+use crate::ffi::InterpreterExtractError;
+use crate::Artichoke;
+
+impl BuildHasher<RandomState> for Artichoke {
+    type Error = Error;
+
+    fn build_hasher(&mut self) -> Result<&RandomState, Self::Error> {
```

Comment:
style nit, I prefer using the associated types here to mirror the trait definition:

```suggestion
    fn build_hasher(&mut self) -> Result<&Self::BuildHasher, Self::Error> {
```